### PR TITLE
notebook related documentation

### LINKED ISSUE #839: specviz load data via API fails
Following instructions one page 5 of jdaviz.readthedocs.io/en/latest/pecviz/import_data.html there seems to
something missing as initiating from inside a juypter notebook: 

specviz = SpecViz()

does not seem to initiate the app window in the notebook, that will load the user defined spec1d  with the

specviz.load_spectrum(spec1d)

the spec1d array is created from reading in a user created ascii file with 3 columns, wavelength, flux, uncertainty in
the usual spectultils fashion.

note that insinuating a jdaviz app window with in the notebook environment does work, but the IMPORT+ fails
when attempting to read the file directly.

Is the code snippets presented in the documentation correct?

### PR DESCRIPTION
**Pull request by orifox**
New documentation specific to working in a Notebook.  I wanted to pull out a lot of information that is currently buried or non-existent into one place.  That means there is a little redundancy.  I don't mind that for the time being.  If these changes look correct, we can worry about how to consolidate all the information and potentially reorganize in the future.

EDIT: Rendered doc at https://jdaviz--832.org.readthedocs.build/en/832/index.html

EDIT: Fix #839

### CODE REVIEW

**Comment by pllim** [comment]
You forgot to include your new page in root level index, please fix. Thanks!

`docs/notebook/index.rst: WARNING: document isn't included in any toctree`

**Comment by pllim** [comment]
Update: I pushed a commit to fix the doc build.

**Comment by pllim** [comment]
For future reference -- FYI:

* You can try to build the doc locally first by running `pip install -e . ; cd docs; make html` to make sure there are no errors/warnings.
* If you want to use a new section reference, you have to create the new tags too. They do not magically create themselves, unlike some other software you might be used to.

File: docs/notebook/export_data.rst [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+***********************************
+
+Specific instructions on exporting data from Jdaviz to your notebook vary slightly for each instance of Jdaviz, including :ref:`specviz-notebook`, :ref:`cubeviz-notebook`, :ref:`mosviz-notebook`, and Imviz.  In all instances, however, you can export data products from the Jdaviz into your notebook and, ultimately, save those variables locally.
+
+Export Position Regions (Imviz)
+-------------------------------
+
+In `Imviz`, you can extract positional regions::
+
+    regions=myinstance.get_interactive_regions()
```
Comment (pllim) [inline docs/notebook/export_data.rst:14]: This comment applies to all similar occurences. Should the example more closely match the actual example notebooks?

```suggestion
    regions = viz.get_interactive_regions()
```

File: docs/notebook/import_data.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+
+For Mosviz::
+
+    from jdaviz.configs.mosviz.helper import MosViz as Mosviz
+    myinstance = Mosviz()
+    myinstance .app
+    import tempfile
+    data_dir = tempfile.gettempdir()
+    mosviz.load_data(directory=data_dir, instrument="nirspec") # Or niriss
```
Comment (pllim) [inline docs/notebook/import_data.rst:44]: This is all fake code anyway, no reason to introduce "tempfile" here to complicate things.

```suggestion
    mosviz.load_data(directory="/path/to/data/file", instrument="nirspec") # Or niriss
```

**Review by pllim — COMMENTED** [review]
Thanks for the documentation! Definitely good to have something. We can polish things up to the standards of Astropy Tutorials in the future.

All the intersphinx stuff is still broken but I guess we can defer that. Seems to be a problem across the board.

Minor comments below.

**Comment by kaya4me2** [comment]
Documentation link still not quite correct ... in order to read in ascii file an initiate the Jdaviz interface had to to the following in the Jupiter notebook code: (i.e., the myinstance.app line placement )

correct_nujdaviz_ascii_call

Note that the flux density axis has no units on the axis, but the spectra looks about right when plotted in the visualizer.

Secondly, why aren't the uncertainties plotted ?

**Comment by javerbukh** [comment]
@kaya4me2 Thank you for your comment! Would you be willing to provide the file you are trying to load for testing purposes?

When you mention the flux density axis, are you referring to the data itself or the plot?

Uncertainties can be plotted by running the follow: `myinstance.app.get_viewer("spectrum-viewer").show_uncertainties()`. Would you prefer uncertainties were plotted by default?

**Comment by orifox** [comment]
Addressed all comments above, except:
1) loading uncertainties
2) Y-axis labeling problems (I noticed this myself with different data sets)  

Created new tickets for both.

This PR is now ready for final review.  Already labeled.

File: docs/notebook/export_data.rst [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+***********************************
+
+Specific instructions on exporting data from Jdaviz to your notebook vary slightly for each instance of Jdaviz, including :ref:`specviz-notebook`, :ref:`cubeviz-notebook`, :ref:`mosviz-notebook`, and Imviz.  In all instances, however, you can export data products from the Jdaviz into your notebook and, ultimately, save those variables locally.
+
+Export Position Regions (Imviz)
+-------------------------------
+
+In `Imviz`, you can extract positional regions::
+
+    regions=myviz.get_interactive_regions()
```
Comment (javerbukh) [inline docs/notebook/export_data.rst:14]:
``` suggestion
    regions = myviz.get_interactive_regions()
```
Minor nitpick.

**Comment by javerbukh** [comment]
@orifox One minor comment, otherwise looks good! Ping me when you commit that change and I'll approve and merge.

**Comment by kaya4me2** [comment]
> @kaya4me2 Thank you for your comment! Would you be willing to provide the file you are trying to load for testing purposes?
> 
> When you mention the flux density axis, are you referring to the data itself or the plot?
> 
> Uncertainties can be plotted by running the follow: `myinstance.app.get_viewer("spectrum-viewer").show_uncertainties()`. Would you prefer uncertainties were plotted by default?

Yes that would be preferred (sorry about the delay in response), also where does the suggest line
myinstance.app.get_viewer("spectrum-viewer").show_uncertainties() come the flow of initiating the spexviz app in the notebook?

**Comment by javerbukh** [comment]
It would come after data has been loaded into the application.

**Review by javerbukh — APPROVED** [review]
Looks good!

**Pull request merged**